File: typescript/base-hmi-types/src/screens/base/HmiFaceplateType.ts
import { HmiScreenBase } from "./HmiScreenBase.js";
import { HmiObjectType } from "./HmiObjectType.js";
import { HmiScreenKind } from "./HmiScreenKind.js";
import { HmiFaceplateInterfaceMember } from "./HmiFaceplateInterfaceMember.js";

export class HmiFaceplateType extends HmiScreenBase {
  version?: string;
  readonly interfaceProperties: HmiFaceplateInterfaceMember[] = [];
  readonly tagInterfaceProperties: HmiFaceplateInterfaceMember[] = [];

  constructor() {
    super();
    this.hmiObjectType = HmiObjectType.HmiFaceplateType;
    this.kind = HmiScreenKind.Faceplate;
  }
}

File: src/BaseHmiTypes/Screens/Base/HmiFaceplateType.cs
namespace BaseHmiTypes.Screens.Base;

public class HmiFaceplateType : HmiScreenBase
{
    public HmiFaceplateType()
    {
        HmiObjectType = BaseHmiTypes.Screens.Base.HmiObjectType.HmiFaceplateType;
        Kind = HmiScreenKind.Faceplate;
    }

    public string? Version { get; set; }

    public IList<HmiFaceplateInterfaceMember> InterfaceProperties { get; } = new List<HmiFaceplateInterfaceMember>();

    public IList<HmiFaceplateInterfaceMember> TagInterfaceProperties { get; } = new List<HmiFaceplateInterfaceMember>();
}
